test/ref_ops_test.cpp

```diff
@@ -2524,6 +2524,82 @@ TEST_CASE(gather_test)
     }
 }
 
+TEST_CASE(gather_dyn_test0)
+{
+    {
+        // Dynamic data, static indices
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        std::vector<int> data(2 * 3);
+        std::iota(data.begin(), data.end(), 0);
```

Comment:
move these definitions of `data` and `indices` below where it is used and passed to create `migraphx::argument` to  make migraphx program contiguous so that it is easier to read.